v2/gcs-to-sourcedb/src/test/java/com/google/cloud/teleport/v2/templates/GCSToSourceDbDatatypeIT.java

```diff
@@ -0,0 +1,288 @@
+            .build();
+    spannerResourceManager.write(m);
+  }
+
+  private List<Throwable> assertionErrors = new ArrayList<>();
+
+  private void assertAll(Runnable... assertions) throws MultipleFailureException {
+    for (Runnable assertion : assertions) {
+      try {
+        assertion.run();
+      } catch (AssertionError e) {
+        assertionErrors.add(e);
+      }
+    }
+    if (!assertionErrors.isEmpty()) {
+      throw new MultipleFailureException(assertionErrors);
+    }
+  }
+
+  private void assertRowInMySQL() throws InterruptedException, MultipleFailureException {
+    long rowCount = 0;
+    for (int i = 0; rowCount != 1 && i < 30; ++i) {
+      rowCount = jdbcResourceManagers.get(0).getRowCount(TABLE1);
+      LOG.info("Row count = {}, Waiting for 10s if row count not = 1", rowCount);
+      Thread.sleep(10000);
+    }
+    assertThat(rowCount).isEqualTo(1);
+    List<Map<String, Object>> rows = jdbcResourceManagers.get(0).readTable(TABLE1);
+    assertThat(rows).hasSize(1);
+    Map<String, Object> row = rows.get(0);
+    assertAll(
+        () -> assertThat(row.get("varchar_column")).isEqualTo("value1"),
+        () -> assertThat(row.get("tinyint_column")).isEqualTo(10),
+        () -> assertThat(row.get("text_column")).isEqualTo("text_column_value"),
+        () -> assertThat(row.get("date_column")).isEqualTo(java.sql.Date.valueOf("2024-05-24")),
+        () -> assertThat(row.get("smallint_column")).isEqualTo(50),
+        () -> assertThat(row.get("mediumint_column")).isEqualTo(1000),
+        () -> assertThat(row.get("int_column")).isEqualTo(50000),
+        () -> assertThat(row.get("bigint_column")).isEqualTo(987654321),
+        () -> assertThat(row.get("float_column")).isEqualTo(45.67f),
+        () -> assertThat(row.get("double_column")).isEqualTo(123.789),
+        () -> assertThat(row.get("decimal_column")).isEqualTo(new BigDecimal("1234.56")),
+        () ->
+            assertThat(row.get("datetime_column"))
+                .isEqualTo(java.time.LocalDateTime.of(2024, 2, 8, 8, 15, 30)),
+        () ->
+            assertThat(row.get("timestamp_column"))
+                .isEqualTo(java.sql.Timestamp.valueOf("2024-02-08 08:15:30.0")),
+        () -> assertThat(row.get("time_column")).isEqualTo(java.sql.Time.valueOf("14:30:00")),
+        () -> assertThat(row.get("year_column")).isEqualTo(java.sql.Date.valueOf("2022-01-01")),
+        () -> assertThat(row.get("char_column")).isEqualTo("char_col"),
+        () -> assertThat(row.get("tinytext_column")).isEqualTo("tinytext_column_value"),
+        () -> assertThat(row.get("mediumtext_column")).isEqualTo("mediumtext_column_value"),
+        () -> assertThat(row.get("longtext_column")).isEqualTo("longtext_column_value"),
+        () ->
+            assertThat(row.get("tinyblob_column"))
+                .isEqualTo("tinyblob_column_value".getBytes(StandardCharsets.UTF_8)),
+        () ->
+            assertThat(row.get("blob_column"))
```

Comment:
Could you add some scenarios for escape characters? There was a bug around this and while it is covered via UT, will be good to add in IT as well. Please refer https://github.com/GoogleCloudPlatform/DataflowTemplates/pull/1665
If not possible, please add a TODO